SystemReference/Payload/Camera/Camera.cpp

```diff
@@ -6,117 +6,304 @@
 
 #include "Fw/Types/BasicTypes.hpp"
 #include <SystemReference/Payload/Camera/Camera.hpp>
+#include <libcamera/libcamera.h>
+#include <errno.h>
+#include <unistd.h>
+#include <sys/mman.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <fstream>
 
 namespace Payload {
 
-// ----------------------------------------------------------------------
-// Construction, initialization, and destruction
-// ----------------------------------------------------------------------
+  // ----------------------------------------------------------------------
+  // Construction, initialization, and destruction
+  // ----------------------------------------------------------------------
 
-Camera ::Camera(const char *const compName)
-    : CameraComponentBase(compName), m_photoCount(0){}
+  Camera ::Camera(const char *const compName): CameraComponentBase(compName), m_photoCount(0){}
 
-void Camera ::init(const NATIVE_INT_TYPE queueDepth,
-                   const NATIVE_INT_TYPE instance) {
-  CameraComponentBase::init(queueDepth, instance);
-}
-
-bool Camera::open(I32 deviceIndex) {
-#ifdef USES_OPENCV
-  if (m_capture.isOpened()) {
-    this->log_ACTIVITY_LO_CameraAlreadyOpen();
-    return true;
+  void Camera ::init(const NATIVE_INT_TYPE queueDepth, const NATIVE_INT_TYPE instance) {
+    CameraComponentBase::init(queueDepth, instance);
+    camManager = std::make_unique<libcamera::CameraManager>();
   }
-  m_capture.open(deviceIndex);
-  if (!m_capture.isOpened()) {
+  // start camera manager and acquire camera
+  // return true on success, false otherwise
+  bool Camera::open(I32 deviceIndex) {
+    int returnCode = camManager->start();
+    // check to see if the camera manager was started successfully
+    if (returnCode == 0) {
+      // if no cameras were detected, emit a CameraNotDetect event
+      if(camManager->cameras().empty()) {
+        this->log_WARNING_HI_CameraNotDetected();
+        return false;
+      }
+      // get the camera from the CameraManager and acquire it
+      m_capture = camManager->cameras()[deviceIndex];
+      returnCode = m_capture->acquire();
+
+      // camera was acquired successfully
+      if (returnCode == 0) {
```

Comment:
Do we need to set resolution on startup?

Reply:
We don't since its taken care of at each capture. Although, I think it would be better if we set it only at open and resolution param update, but I haven't had any luck with that.

Reply:
Make CameraManager a local var here if its not used anywhere